sdk/storage/storage-blob/rollup.base.config.js

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import nodeResolve from "rollup-plugin-node-resolve";
+import multiEntry from "rollup-plugin-multi-entry";
+import cjs from "rollup-plugin-commonjs";
+import json from "rollup-plugin-json";
+import replace from "rollup-plugin-replace";
+import { uglify } from "rollup-plugin-uglify";
+import sourcemaps from "rollup-plugin-sourcemaps";
+import shim from "rollup-plugin-shim";
+// import visualizer from "rollup-plugin-visualizer";
+
+const version = require("./package.json").version;
+const banner = [
+  "/*!",
+  ` * Azure Storage SDK for JavaScript - Blob, ${version}`,
+  " * Copyright (c) Microsoft and contributors. All rights reserved.",
+  " */"
+].join("\n");
+
+const pkg = require("./package.json");
+const depNames = Object.keys(pkg.dependencies);
+const production = process.env.NODE_ENV === "production";
+
+export function nodeConfig(test = false) {
+  const externalNodeBuiltins = [
+    "@azure/ms-rest-js",
+    "crypto",
+    "fs",
+    "events",
+    "os",
+    "stream"
+  ];
+  const baseConfig = {
+    input: "dist-esm/src/index.js",
+    external: depNames.concat(externalNodeBuiltins),
+    output: {
+      file: "dist/index.js",
+      format: "cjs",
+      sourcemap: true
+    },
+    plugins: [
+      sourcemaps(),
```

Comment:
What does "sourcemaps()" do? I remembered a sourcemap file is generated under dist folder without this plugin.

Reply:
This plugin helps in grabbing source maps from sourceMappingURLs.

For instance, look at the differences in nyc output.
- with `sourcemaps()`
![image](https://user-images.githubusercontent.com/10452642/57413111-df385980-71a7-11e9-9b9b-33eefcde995f.png)

- without `sourcemaps()`
![image](https://user-images.githubusercontent.com/10452642/57413128-e8c1c180-71a7-11e9-8249-19bc5770bd1e.png)

Reply:
Got it. Can you remove "dotenv" from coverage report?

Reply:
sourcemaps is for threading the sourcemap across multiple transformations. Rollup by default won't consume sourcemaps so the `dist` sourcemaps would be a mapping to `dist-esm` rather than the original TS code.

Reply:
>Got it. Can you remove "dotenv" from coverage report?

Update -

I tried excluding that specific file(and every possible node_modules path) by adding rules in `.nycrc` file and could not exclude this file.

 Looks like `nyc` is not properly honouring the `exclude` rule in `.nycrc`. I see some similar recent user issues in the nyc repo - https://github.com/istanbuljs/nyc/issues/1099 

 I'm opening a new issue to investigate further - https://github.com/Azure/azure-sdk-for-js/issues/2820

Reply:
@HarshaNalluru - I apologize.  The issue 1099 that you reference was mine.  It was user error.